app/src/main/java/org/thoughtcrime/securesms/conversation/v2/components/AlbumThumbnailView.kt

```diff
@@ -76,10 +79,14 @@ class AlbumThumbnailView : FrameLayout {
                 val slide = slides.getOrNull(index) ?: return
                 // only open to downloaded images
                 if (slide.transferState == AttachmentTransferProgress.TRANSFER_PROGRESS_FAILED) {
-                    // restart download here
+                    // Restart download here (on IO thread)
                     (slide.asAttachment() as? DatabaseAttachment)?.let { attachment ->
                         val attachmentId = attachment.attachmentId.rowId
-                        JobQueue.shared.add(AttachmentDownloadJob(attachmentId, mms.getId()))
```

Comment:
Any reason we are adding this from the coroutine scope compared to just doing it normally?

Reply:
Yea so the flow is `JobQueue.shared.add -> addWithoutExecuting -> MessagingModuleConfiguration.shared.storage.persistJob(job)`, this then triggers a database `insertOrUpdate` call which is blocking if another database write is in process resulting in the main thread hanging.

I haven't been able to cause this one to hang (mostly because by the time I have a failed attachment download on my test account most of the heavy database operations are already completed) but I figured it was safer to add it (also means it's consistent with the `VisibleMessageContentView` behaviour which I did see causing the main thread to hang when opening a conversation while lots of message insertion was going on on my test account)